p2p/src/utils/mux.rs

```diff
@@ -78,6 +84,32 @@ impl<E: Spawner, S: Sender, R: Receiver> Muxer<E, S, R> {
         (mux, handle)
     }
 
+    /// Like [Muxer::new], but also registers a backup channel that can be used to handle messages
+    /// from unregistered subchannels.
+    #[allow(clippy::type_complexity)]
+    pub fn new_with_backup(
+        context: E,
+        sender: S,
+        receiver: R,
+        mailbox_size: usize,
+    ) -> (
+        Self,
+        MuxHandle<S, R>,
+        mpsc::Receiver<BackupResponse<R::PublicKey>>,
+    ) {
+        let (mut mux, handle) = Self::new(context, sender, receiver, mailbox_size);
+
+        let (tx, rx) = mpsc::channel(mailbox_size);
+        mux.backup = Some(tx);
+
+        (mux, handle, rx)
+    }
+
+    /// Returns an [AnySubSender] for sending messages over any subchannel.
+    pub fn any_subsender(&self) -> AnySubSender<S> {
```

Comment:
It turned out to be a good idea to not assume wanting to send to anyone also means you want a backup (in boundary handling we listen on "pending" but respond on "recovered").

Reply:
I added a typestate builder pattern that I think cleans this up a bit.